cmd/provider/main.go

```diff
@@ -144,6 +148,23 @@ func main() {
 
 	log := logging.NewLogrLogger(zl.WithName("provider-ceph"))
 
+	// Init otel tracer provider
+	tp, err := traces.InitTracerProvider(*metricsExportAddress, *metricsExportTimeout, *metricsExportInterval)
+	if err != nil {
+		log.Debug("error starting tracer provider", "error", err.Error())
+	}
+	if tp != nil {
+		defer func() {
+			ctx, cancel := context.WithTimeout(context.Background(), *metricsExportTimeout)
+			defer cancel()
+
+			// flush
+			if err := tp.Shutdown(ctx); err != nil {
+				log.Debug("failed to gracefully shutdown tracer provider", "error", err.Error())
+			}
+		}()
+	}
```

Comment:
Could we have this as "opt-in" with a flag, something like `--enable-open-telemetry-tracing` ? not sure if it's something that should be forced upon every user by default.